src/cript/api/api.py

```diff
@@ -429,6 +470,126 @@ def save(self, project: Project) -> None:
 
+    def upload_file(self, file_path: Union[Path, str]) -> str:
+        """
+        uploads a file to AWS S3 bucket and returns a URL of the uploaded file in AWS S3
+        The URL is has no expiration time limit and is available forever
+
+        1. take an  file path of type path or str to the file on local storage
+            * see Example for more details
+        1. convert the file path to pathlib object, so it is versatile and
+            always uniform regardless if the user passes in a str or path object
+        1. get the file
+        1. rename the file to avoid clash or overwriting of previously uploaded files
+            * change file name to `original_name_uuid4.extension`
+                *  `document_42926a201a624fdba0fd6271defc9e88.txt`
+        # it would be without the hyphens like: `document_42926a201a624fdba0fd6271defc9e88.txt`
+        1. upload file to AWS S3
+        1. get the link of the uploaded file and return it
+
+
+        Examples
+        --------
+        ```python
+        import cript
+
+        api = cript.API(host, token)
+
+        # programmatically create the absolute path of your file, so the program always works correctly
+        my_file_path = (Path(__file__) / Path('../upload_files/my_file.txt')).resolve()
+
+        my_file_s3_url = api.upload_file(absolute_file_path=my_file_path)
+        ```
+
+        Parameters
+        ----------
+        file_path: Union[str, Path]
+            file path as str or Path object. Path Object is recommended
+
+        Raises
+        ------
+        FileNotFoundError
+            In case the file could not be found because the file does not exist
+
+        Returns
+        -------
+        url: str
+            url of the AWS S3 uploaded file to be put into the File node source attribute
+        """
+
+        # convert file path from whatever the user passed in to a pathlib object
+        file_path = pathlib.Path(file_path).resolve()
+
+        # get file_name and file_extension from absolute file path
+        # file_extension includes the dot, e.g. ".txt"
+        file_name, file_extension = os.path.splitext(os.path.basename(file_path))
+
+        # generate a UUID4 string without dashes, making a cleaner file name
+        uuid_str = str(uuid.uuid4().hex)
+
+        new_file_name: str = f"{file_name}_{uuid_str}{file_extension}"
+
+        # e.g. "directory/file_name_uuid.extension"
+        object_name = f"{self._BUCKET_DIRECTORY_NAME}/{new_file_name}"
+
+        # upload file to AWS S3
+        self._s3_client.upload_file(file_path, self._BUCKET_NAME, object_name)
+
+        # Generate a presigned URL to access the file from S3 that is available forever
+        s3_file_url = self._s3_client.generate_presigned_url(
+            "get_object",
+            Params={"Bucket": self._BUCKET_NAME, "Key": object_name},
+        )
+
+        return s3_file_url
+
+    def download_file(self, file_url: str, destination_path: Union[Path, str]) -> None:
```

Comment:
also, should be user facing.

What should be user facing though is a spot where all files are downloaded to.
I would say, we initially have a default either `.` or `~/Downloads` but users can set with the API class or an env variable where all files should be stored that are downloaded.

Reply:
I was not sure about `~/Downloads` because I wasn't sure if that would work with all OS

I can put it within the cwd